frontend/src/components/WorkflowDashboard.vue

```diff
@@ -56,7 +56,14 @@ export default {
           this.showHealthyBuilds ? true : !this.isIdleHealthyBuild(lastBuildStatus, activity))
         .filter(({ triggeredEvent }) =>
           !this.hasPreferredTriggeredEvents ||
-            preferences.showBuildsDueToTriggeredEvents.indexOf(triggeredEvent) !== -1);
+            preferences.showBuildsDueToTriggeredEvents.indexOf(triggeredEvent) !== -1)
+        .map((data) => {
+          const parts = data.name.split(' :: ');
+          return {
+            ...data,
+            name: preferences.showJobNameOnly ? parts[parts.length - 1] : data.name
+          };
```

Comment:
It'll be better to pull out this as mapper. Depends on the configuration use required mapper.

Something  like:

```
const jobNameMapper = (data) => {
    const parts = data.name.split(' :: ');
    return {
        ...data,
        name: parts[parts.length - 1]
    }
}

const identityMapper = data => data;

const mapper = preferences.showJobNameOnly ? jobNameMapper : identityMapper;

```